src/wallet/wallet.cpp

```diff
@@ -3979,20 +3996,25 @@ std::shared_ptr<CWallet> CWallet::CreateWalletFromFile(interfaces::Chain& chain,
 
     if (fFirstRun)
     {
-        // ensure this wallet.dat can only be opened by clients supporting HD with chain split and expects no default key
-        walletInstance->SetMinVersion(FEATURE_LATEST);
+        // ensure this wallet can only be opened by clients supporting HD with chain split and expects no default key
+        if (!create_empty) {
+            walletInstance->SetMinVersion(FEATURE_PRE_SPLIT_KEYPOOL);
```

Comment:
@Empact in an earlier version I added a new feature version for this, but then I realized it wasn't necessary. Forgot to remove this bit. Fixed.

Reply:
I think instead of using a version number you should use a wallet flag that is required. The flag can be unset once the wallet is no longer empty. This would remove the need for a new version number while still preventing older software from opening the wallet.